tools/SourceKit/tools/sourcekitd/lib/API/sourcekitdAPI-InProc.cpp

```diff
@@ -0,0 +1,941 @@
+
+  sourcekitd_variant_type_t getVariantType() override {
+    return SOURCEKITD_VARIANT_TYPE_INT64;
+  }
+
+  Optional<int64_t> getInt64() override {
+    return Optional<int64_t>(Storage);
+  }
+
+  static bool classof(const SKDObject *O) {
+    return O->getKind() == OK_Int64;
+  }
+private:
+  int64_t Storage;
+};
+
+class SKDUID: public SKDObject {
+public:
+  SKDUID(sourcekitd_uid_t Value) : SKDObject(OK_UID), Storage(Value) {}
+
+  sourcekitd_variant_type_t getVariantType() override {
+    return SOURCEKITD_VARIANT_TYPE_UID;
+  }
+
+  sourcekitd_uid_t getUID() override {
+    return Storage;
+  }
+
+  static bool classof(const SKDObject *O) {
+    return O->getKind() == OK_UID;
+  }
+private:
+  sourcekitd_uid_t Storage;
+};
+
+class SKDBool: public SKDObject {
+public:
+  SKDBool(bool Value) : SKDObject(OK_Bool), Storage(Value) {}
+
+  sourcekitd_variant_type_t getVariantType() override {
+    return SOURCEKITD_VARIANT_TYPE_BOOL;
+  }
+
+  bool getBool() override {
+    return Storage;
+  }
+
+  static bool classof(const SKDObject *O) {
+    return O->getKind() == OK_Bool;
+  }
+private:
+  bool Storage;
+};
+
+class SKDCustomData: public SKDObject {
+public:
+  SKDCustomData(CustomBufferKind BufferKind, 
+                std::unique_ptr<llvm::MemoryBuffer>& MemBuf)
+  : SKDObject(OK_CustomData), BufferKind(BufferKind),
+    BufferPtr(llvm::MemoryBuffer::getMemBufferCopy(
+                                                MemBuf->getBuffer(), 
+                                                MemBuf->getBufferIdentifier())) 
+    {}
+
+  SKDCustomData(SKDCustomData const&) = delete;
+  SKDCustomData &operator=(SKDCustomData const&) = delete;
+
+  sourcekitd_variant_type_t getVariantType() override {
+    switch (BufferKind) {
+      case CustomBufferKind::TokenAnnotationsArray:
+      case CustomBufferKind::DocSupportAnnotationArray:
+      case CustomBufferKind::CodeCompletionResultsArray:
+        return SOURCEKITD_VARIANT_TYPE_ARRAY;
+    }
+    llvm::report_fatal_error("sourcekitd object did not resolve to a known type");
+  }
+
+  CustomBufferKind getBufferKind() {
+    return BufferKind;
+  }
+
+  const void *getDataPtr() {
+    return BufferPtr->getBuffer().data();
+  }
+
+  static bool classof(const SKDObject *O) {
+    return O->getKind() == OK_CustomData;
+  }
+private:
+  CustomBufferKind BufferKind;
+  std::unique_ptr<llvm::MemoryBuffer> BufferPtr;
+};
+
+class SKDError: public SKDObject {
+public:
+  SKDError(sourcekitd_error_t ErrorKind, StringRef Description)
+  : SKDObject(OK_Error), ErrorKind(ErrorKind), Description(Description.str()) {}
+
+  SKDError(SKDError const&) = delete;
+  SKDError &operator=(SKDError const&) = delete;
+
+  sourcekitd_variant_type_t getVariantType() override {
+    llvm::report_fatal_error("sourcekitd error object is not a variant type");
+  }
+
+  sourcekitd_error_t getErrorKind() {
+    return ErrorKind;
+  }
+
+  StringRef getDescription() {
+    return StringRef(Description);
```

Comment:
I suggest making this `const std::string &getDescription() const`, it will be useful later on.